submit-web/src/components/SubmissionItem/ManagementPlanSubmission/ManagementPlanProponentView/DocumentUploadSection.tsx

```diff
@@ -53,6 +53,17 @@ export const DocumentUploadSection = () => {
   }, [submissionItem, getDocumentSubmissions, initializeFiles]);
 
   const handleOnDrop = (acceptedFiles: File[], folder: string) => {
+    const isManagementPlanFolder =
```

Comment:
when we add a file we do two things:
1- we call addPendingFile which updates the file store
2- we call setValue on the schema.
![image](https://github.com/user-attachments/assets/031ff999-3088-47ec-b45f-4119e83ca115)

These two should behave together. Find a way to pass a prop of max length to the FileUpload so the on Drop and the setValue are not called from the start

Reply:
Refactored code to pass and utilize maxFiles